### DISCUSSION TITLE
Tengo muchas dudas de la kata 10

### DISCUSSION THREAD

**Discussion by JaelDS** [LaunchX-InnovaccionVirtual/CursoIntroPython#223] [Q&A] [answered]
¿Todo es en el cmd o qué parte es en un ipyn y qué parte en el cmd?
¿La definición de la función es en el cmd o en el ipyn?
¿Cómo hago para que el código funcione y controle los errores?
No le estoy encontrando el modo :(

**Comment by htmtmx** [comment]
En mi caso solo ejecute la primer línea de **open("/path/to/mars.jpg")**, todo lo demás lo maneje con el archivo config.py
En VisualStudioCode viene el botón para ejecutarlo y en la parte inferior cuando lo ejecutas te sale una pequeña consola

![image](https://user-images.githubusercontent.com/69724879/153774771-71f2e3ad-ab2a-4ac8-92c3-9e83ad5485a1.png)


Todo lo base en screenshots y las uní en un pdf

**Reply by htmtmx** [reply]
Como tal quiero pensar que la kata no necesariamente debe salirnos al pie de la letra, puesto que hay partes que tienen mala traducción o faltan palabras, aparte de que al final dice, **Intenta realizar los ejercicios descritos en este documento.**

No como en las demás donde si te pide que lo hagas

**Reply by SeleneMancilla** [reply]
Aquí está el material original por si tienen algunas dudas en cuanto a la traducción
https://docs.microsoft.com/en-us/learn/modules/python-error-handling/

**Reply by AndresAVargas** [reply]
pero sustituyes todo el codigo del open.py? osea borras la funcion main?

**Reply by AndresAVargas** [reply]
en el articulo que compartieron dice algo de eliminar el archivo config.txt, en que momento creamos ese archivo y donde lo quieto?

**Comment by doguedogue** [comment]
Hola, con los notebooks es más fácil ya que los vas probando (ipnb y ejecutarlo en el mismo notebook)
Pero también los puedes hacer como un programa standalone python (extensión py y ejecutarlo en cmd)

**Comment by AndreCerva** [comment, marked as answer]
Hola @JaelDS, tranquilo, no te preocupes si te sientes confundido, tu puedes!
1) Podrías hacer tus programas tanto con extensión .py (programa de Python normal) y ejecutarlo en el cmd  o en la terminal que ofrece vs code, con la siguiente instrucción: "python nombre.py" (sin comillas), con ello verás las excepciones que se provocan dentro del código debido a los errores, o en un .ipynb (Un Jupyter Notebook) y ejecutarlo por línea y de igual forma ahí te mostrará los tracebacks, es a tu elección, lo que se te haga más cómodo de trabajar, Fer comentó que por esa parte no habría problema.
2) En el .ipynb o .py
3) Para hacer funcionar el código y manejar los errores, tienes que seguir la sintaxis de este siguiente ejemplo:
...     try: #(intenta hacer lo que se encuentra en las siguientes líneas indexadas)
...         x = int(input("Please enter a number: ")) #(el programa al ejecutarse intentará pedir un numero al usuario por consola)
...     except ValueError: #(En caso de que se introduzca un carácter que no es un número, por ejemplo, una letra o carácter especial, etc ->
                                    # -> se retornará el traceback: ValueError, dado a que se requiere un int, para tener control sobre esto y no se ->
                                    #-> termine de ejecutar nuestro programa con el error, con el except lo que estamos diciendo es que en caso de >
                                    #-> se presente este caso que te comento, entonces haga la siguiente linea de código que se tiene indexada)
...         print("Oops!  That was no valid number.  Try again...")#(Si se tiene el traceback de ValueError, entonces mostrará este mensaje)

Bajo ese principio, solo habría que seguir paso a paso la documentación que Fer creo para este modulo 10, si aún no lo tienes del todo claro o te surgen más dudas, no te preocupes, sigue intentándolo y no dudes en preguntar cualquier duda:)

**Reply by JaelDS** [reply]
mi mayor duda es cómo ejecutar en los notebooks la parte que tiene $ python "nombre del archivo"

**Reply by CeriOwl** [reply]
Esto lo hacen en el documento porque lo están haciendo por archivos python (los .py) en vez de hacerlo por notebooks como lo estamos haciendo nosotros en el curso. [Link para mas información de los archivos .py](https://www.mclibre.org/consultar/python/otros/python-uso.html)
Es por esto que si quisieras hacerlo en notebooks, tómalo como si cada que creas una parte para escribir código fuera un archivo python por separado, entonces cada que lo ejecutas es como si hicieras lo de **$ python "nombre del archivo"**.

**Reply by JaelDS** [reply]
Muchas gracias, ahorita lo que no estoy logrando es generar el error de directorio del archivo .py. Para comprobar el error podría hacer un open() ¿no?

**Reply by CeriOwl** [reply]
Exactamente, esa es la forma en la que hay que comprobar el error 👍 
Suerte con este último tramo de las katas 😆

**Reply by JaelDS** [reply]
Me da un error de Permission denied, no el del directory :(

**Reply by MarioDeLaTorreR** [reply]
> Me da un error de Permission denied, no el del directory :(

me pasa lo mismo pero si lo estoy haciendo en un notebook, si lo hago en un archivo.py me marca otra cosa diferente pero no lo que debe salir :/

**Reply by AndresAVargas** [reply]
lograron algo? :c

**Reply by AndreCerva** [reply]
Hola, yo les recomendaría que si siguen estancados con esta kata 10 y por ello aún no han respondido el formulario para continuar con Front End envíen ya así su kata y después lo pueden retomar, para que no se atrasen en Front End. Y para el error que comentan, si pudieran subir una foto de su código y el error que les da se los agradecería, o si lo prefieren, me pueden enviar un mensaje privado con su código y el error por [ig](https://www.instagram.com/andrscerva/) y con mucho gusto resolvemos juntos el problema
